RecoTracker/LSTCore/src/alpaka/NeuralNetwork.h

```diff
@@ -70,58 +70,42 @@ namespace lst::t5dnn {
-    bool is_endcap3 = (modulesInGPU.subdets[lowerModuleIndex3] == 4);  // true if anchor hit 3 is in the endcap
-    bool is_endcap4 = (modulesInGPU.subdets[lowerModuleIndex4] == 4);  // true if anchor hit 4 is in the endcap
-    bool is_endcap5 = (modulesInGPU.subdets[lowerModuleIndex5] == 4);  // true if anchor hit 5 is in the endcap
 
     float t5_eta = mdsInGPU.anchorEta[md_idx_for_t5_eta_phi];
 
     // Constants
-    constexpr unsigned int kinputFeatures = 24;
+    constexpr unsigned int kinputFeatures = 18;
     constexpr unsigned int khiddenFeatures = 32;
 
     // Build DNN input vector (corresponding output N-tuple branch noted in parenthetical in comment)
     float x[kinputFeatures] = {
-        mdsInGPU.anchorEta[mdIndex1],                                    // inner T3 anchor hit 1 eta (t3_0_eta)
-        mdsInGPU.anchorZ[mdIndex1] / kZ_max,                             // inner T3 anchor hit 1 z (t3_0_z)
-        alpaka::math::sqrt(acc, x1 * x1 + y1 * y1) / kR_max,             // inner T3 anchor hit 1 r (t3_0_r)
-        float(modulesInGPU.layers[lowerModuleIndex1] + 6 * is_endcap1),  // inner T3 anchor hit 1 layer (t3_0_layer)
-        mdsInGPU.anchorEta[mdIndex2],                                    // inner T3 anchor hit 2 eta (t3_2_eta)
-        mdsInGPU.anchorZ[mdIndex2] / kZ_max,                             // inner T3 anchor hit 2 z (t3_2_z)
-        alpaka::math::sqrt(acc, x2 * x2 + y2 * y2) / kR_max,             // inner T3 anchor hit 2 r (t3_2_r)
-        float(modulesInGPU.layers[lowerModuleIndex2] + 6 * is_endcap2),  // inner T3 anchor hit 2 layer (t3_2_layer)
-        mdsInGPU.anchorEta[mdIndex3],                                    // inner T3 anchor hit 3 eta (t3_4_eta)
-        mdsInGPU.anchorZ[mdIndex3] / kZ_max,                             // inner T3 anchor hit 3 z (t3_4_z)
-        alpaka::math::sqrt(acc, x3 * x3 + y3 * y3) / kR_max,             // inner T3 anchor hit 3 r (t3_4_r)
-        float(modulesInGPU.layers[lowerModuleIndex3] + 6 * is_endcap3),  // inner T3 anchor hit 3 layer (t3_4_layer)
-        mdsInGPU.anchorEta[mdIndex4],                                    // outer T3 anchor hit 4 eta (t3_2_eta)
-        mdsInGPU.anchorZ[mdIndex4] / kZ_max,                             // outer T3 anchor hit 4 z (t3_2_z)
-        alpaka::math::sqrt(acc, x4 * x4 + y4 * y4) / kR_max,             // outer T3 anchor hit 4 r (t3_2_r)
-        float(modulesInGPU.layers[lowerModuleIndex4] + 6 * is_endcap4),  // outer T3 anchor hit 4 layer (t3_2_layer)
-        mdsInGPU.anchorEta[mdIndex5],                                    // outer T3 anchor hit 5 eta (t3_4_eta)
-        mdsInGPU.anchorZ[mdIndex5] / kZ_max,                             // outer T3 anchor hit 5 z (t3_4_z)
-        alpaka::math::sqrt(acc, x5 * x5 + y5 * y5) / kR_max,             // outer T3 anchor hit 5 r (t3_4_r)
-        float(modulesInGPU.layers[lowerModuleIndex5] + 6 * is_endcap5),  // outer T3 anchor hit 5 layer (t3_4_layer)
```

Comment:
are these category-like variables (layers) not useful anymore? Were these variables significant before the bug?
I guess it's somewhat redundant if hit r and z are in the inputs as well and the layers stood in place as a bug mediator before the bugfix.
Do I understand correctly?

Reply:
The bug in the training file meant that all hit-dependent variables (so most variables in the DNN) were being randomized during training. This is why many variables had near-zero feature importance despite being correlated with variables that had high importance (e.g. the eta values of the 5 hits showed zero importance despite being correlated with "t5_eta" which was being used by the model). After fixing the bug, the layer variables still had 0 importance, so I decided to remove them. I figured like you said that they are redundant if you are giving the model r, z, and eta of every hit already.